### PR TITLE
FreeBSD fixes.

### PR DESCRIPTION
**Pull request by trasz**
Random fixes that should help in making it build on FreeBSD.

### CODE REVIEW

File: src/configure.in [hunk 1]
```diff
@@ -423,7 +423,7 @@ else
   AC_MSG_RESULT(not specified)
 fi
 
-AC_PROG_CC([$RTSCC gcc])
+AC_PROG_CC([$RTSCC gcc clang])
 AC_PROG_CPP()
```
Comment (jepler) [inline src/configure.in]: Is a change unneeded from here because the default AC_PROG_CPP includes clang++ in its list?

**Comment by jepler** [comment]
Unfortunately, I don't think we can accept 'Don't include <sys/fsuid.h>' cc89afd569f08728a2afb98feffd9234d9edbe2a as-is.

When we merge #276 (which tbh I thought I'd previously merged, but let's go head and PR it) we will go back to calling `setfsuid` on Linux, because calling `seteuid` caused missed realtime deadlines on PREEMPT-RT and hard locks on RTAI.

**Comment by jepler** [comment]
I did go ahead and cherry-pick da8b0db9bbc0615259b89e8e64cd8ac0cd1298d8 as 6a3b4a2263 though.  Thanks!

**Comment by trasz** [comment]
Hm, ok; I'll make the fsuid stuff conditional on __linux__ then.  Thanks!

**Comment by jepler** [comment]
OK, I'll close this one up then.  Thanks!

**Pull request closed without merging**